e2e/e2e_test.go

```diff
@@ -87,17 +108,202 @@ var _ = Describe("Whereabouts functionality", func() {
+					replicaCount,
+					podTierLabel(rsName),
+					podNetworkSelectionElements(testNetworkName),
+				)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				By("deleting replicaset with whereabouts net-attach-def")
+				Expect(clientInfo.deleteReplicaSet(replicaSet)).To(Succeed())
+			})
+
+			It("allocates each IP pool entry with a unique pod IP", func() {
+				By("creating max number of pods and checking IP Pool validity")
+
+				for i := 0; i < envVars.numThrashIter; i++ {
+					dt = time.Now()
+					fmt.Printf("iteration %d of %d, %s\n", i, envVars.numThrashIter, dt.String())
+
+					// Set to 0 replicas
+					replicaCount = int32(0)
+					replicaSet, err = clientInfo.updateReplicaSet(replicaSetObject(
+						replicaCount,
+						podTierLabel(rsName),
+						podNetworkSelectionElements(testNetworkName),
+					))
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(WaitForReplicaSetSteadyState(clientInfo.Client, testNamespace, label, replicaSet, rsSteadyTimeout)).To(Succeed())
+
+					// Check for 0 IP pool allocations
+					ipPools, err := k8sClient.ListIPPools(context.Background())
+					Expect(err).To(BeNil())
+					Expect(ipPools[0].Allocations()).To(BeEmpty()) // get ipPool by name rather than by hoping index is right - TODO
+
+					// Increase to desired number of replicas for test
+					allPods, err := clientInfo.Client.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					maxReplicas := (envVars.numComputeNodes*maxPodsPerNode - (len(allPods.Items))) * envVars.fillPercentCapacity / 100
+					replicaCount = int32(maxReplicas)
+					replicaSet, err = clientInfo.updateReplicaSet(replicaSetObject(
+						replicaCount,
+						podTierLabel(rsName),
+						podNetworkSelectionElements(testNetworkName),
+					))
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(WaitForReplicaSetSteadyState(clientInfo.Client, testNamespace, label, replicaSet, rsSteadyTimeout)).To(Succeed())
+
+					Expect(iPPoolConsistency(envVars.kubeconfigPath, k8sClient, clientInfo.Client)).To(Succeed())
+				}
+			})
+		})
 	})
 })
 
-func clusterConfig() (*rest.Config, error) {
+func iPPoolConsistency(kubeconfigPath string, k8sClient *kubeClient.Client, cs *kubernetes.Clientset) error {
+	By("checking if there are any stale IPs in IP pool or any IPs in IP pool that are not seen attached to a pod")
+
+	By("Forcing reconciliation of the cluster...")
+	if err := runIpReconciler(kubeconfigPath); err != nil {
+		return err
+	}
+
+	By("Comparing pod ip assignments to ippool reservations")
+	// TODO - stop this ipPools[0] stuff and get it by name if there's a way to do it. low priority though
+	ipPools, err := k8sClient.ListIPPools(context.Background())
+	if err != nil {
+		return err
+	}
+
+	ipPoolAllocations := ipPools[0].Allocations() // todo, get this more elegantly than ipPools[0]
+	podList, err := ListPods(cs, testNamespace, wbLabelEqual)
```

Comment:
You can get the correct pol by name, rather than listing all, then grabbing the first (and only ...) one.

```golang
poolName := normalizeIPRange(ipv4TestRange)
ipPool, err := k8sClient.GetIPPool(context.Background(), poolName)
```

The thing is the function `GetIPPool` does not exist yet, you'd have to create it :) 

It shouldn't be too hard; just copy https://github.com/k8snetworkplumbingwg/whereabouts/blob/3049dcec3f6a6bf90f2587e8e3a7757574fd1c49/pkg/storage/kubernetes/client.go#L79 , but using `i.client.Get(...)` instead.

You can handle that in a follow-up PR if you prefer, since this test is only using a single pool.